helix-lsp/src/lib.rs

```diff
@@ -204,6 +204,17 @@ pub mod util {
         // in reverse order.
         edits.sort_unstable_by_key(|edit| edit.range.start);
 
+        // If the edit is a full document replacement, then we generate a diff.
+        let is_document_replacement = matches!(edits.first().and_then(|edit| {
+            let start = lsp_pos_to_pos(doc, edit.range.start, offset_encoding)?;
+            let end = lsp_pos_to_pos(doc, edit.range.end, offset_encoding)?;
+            Some(start..end)
+        }), Some(range) if range == (0..doc.len_chars()));
```

Comment:
It seems wasteful to precompute this if edits.len() isn't 1, so I'd put this in a nested if

Reply:
Fixed.